Lesson21

askrynnikov opened the pull request:
- Реализовать инлайн-редактирование названия маршрутов и поездов.
- При создании вагона, сделать выпадающий список с типами вагонов. При выборе типа вагонов с списке, форма динамически изменяется для данного типа вагонов (появляются или исчезают необходимые поля). Подсказка: проще всего это реализовать, имея на странице скрытые формы для всех типов вагонов, а при имзенении значения в выпадающем списке (событие change: api.jquery.com) показывать ту или иную форму.

askrynnikov committed:
Lesson21 part one

 Реализовать инлайн-редактирование названия маршрутов и поездов.

askrynnikov committed:
Lesson21 part second

 При создании вагона, сделать выпадающий список с типами вагонов. При выборе типа вагонов с списке, форма динамически изменяется для данного типа вагонов (появляются или исчезают необходимые поля).

psylone commented on app/assets/javascripts/cars.js:
```diff
@@ -0,0 +1,49 @@
+            $('.top_places').removeClass('hidden');
+            $('.lower_side_places').addClass('hidden');
+            $('.top_side_places').addClass('hidden');
+            $('.seat_places').addClass('hidden');
```
В селектор можно передавать сразу несколько классов: `$('.seat_places, .top_side_places').addClass('hidden')`.

psylone replied:
Можно было пойти другим путём: у самих HTML элементов определить классы с типами вагонов, тогда здесь можно было бы сделать что-то вроде этого:

```javascript
$('.seats').addClass('hidden')
$('.seats ' + $('.type-car select').val()).removeClass('hidden')
```

И появятся все элементы у которых есть класс с текущим значением `$('.type-car select')`.

psylone commented on app/assets/javascripts/trains.js:
```diff
@@ -0,0 +1,25 @@
+$(document).ready(function () {
+    $('a.edit_train').click(function (e) {
+        e.preventDefault();
```
Можно определить эту функцию отдельно и передавать в неё необходимые аргументы: `train_id` и прочее, поскольку содержание этой функции-обработчика и той что для маршрутов очень похожи.

psylone commented on app/views/admin/cars/_form.html.erb:
```diff
@@ -12,35 +12,35 @@
     </div>
 
     <% if @type== "Car" || %w(CompartmentCar EconomyCar SleepingCar).include?(@type) %>
```
Тут бы конечно через объект вагона работать. Например реализовать такой метод: `car.has_seats?(:lower_places)`.

askrynnikov committed:
Lesson21 add car.has_seats?

 Тут бы конечно через объект вагона работать. Например реализовать такой метод: car.has_seats?(:lower_places).
